### PR TITLE
Fix webtoon under-scale

### LINKED ISSUE #394: Ability to Zoom-out in Webtoon Mode
### Description
Hi,

I recently started using Kotatsu, and I'm really enjoying reading manga and webtoons on the app. However, I'm facing a small issue. I noticed that there is no option to zoom out in webtoon mode. It would be great if the app could allow us to see a larger portion of the webtoon on the screen without constantly scrolling.

### Proposal
I thought it would be helpful to have a zoom-out feature in webtoon mode. It would be amazing if the developers could add a setting in the app that lets us zoom out while reading webtoons. This way, we wouldn't have to scroll as much and could view more of the webtoon at once.

### Benefits
Here are some reasons why I think this feature would be beneficial:
1. Improved reading experience: Being able to zoom out would make it easier to read the webtoons, as we could see a larger part of the panel at once.
2. Reduced scrolling: With the option to zoom out, we wouldn't have to constantly scroll to view the entire webtoon.

### Implementation Details
To make it user-friendly, it would be great if the zoom-out option could be added in the app's settings. Specifically, a toggle switch under "Reader settings" for enabling zoom-out in webtoon mode would be really helpful.

### Additional Notes
As a user of Kotatsu, I believe that having a zoom-out option in webtoon mode would greatly enhance my reading experience. Thank you for considering this feature request. I appreciate all the hard work you do to make Kotatsu an amazing app for manga and webtoon lovers like me!

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.

### PR DESCRIPTION
**Pull request by VietAnh14**
Close #394
This fix still have a bug at an edge case, but it's not a big bug and I think it's good enough to use.
For the zoom out setting, I think the current webtoon zoom  setting are enough so I don't implement it.

### CODE REVIEW

**Comment by VietAnh14** [comment]
This is the bug that I mentioned, and it doesn't often happen

https://github.com/KotatsuApp/Kotatsu/assets/37103340/ac9d2426-87b7-4a32-9f02-03648b7ff2d1

File: app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt [hunk 1]
```diff
@@ -98,6 +111,15 @@ class WebtoonRecyclerView @JvmOverloads constructor(
 		listeners.forEach { it.dispatchScroll(this, dy, centerPosition) }
 	}
 
+	fun relayoutChildren() {
+		forEach { child ->
+			(child as WebtoonFrameLayout).target.requestLayout()
+		}
+		detachedViews.forEach {child ->
+			(child as WebtoonFrameLayout).target.requestLayout()
```
Comment (Koitharu) [inline app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt:119]: what reason to request layout of detached view?

File: app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt [hunk 2]
```diff
@@ -12,6 +14,17 @@ class WebtoonRecyclerView @JvmOverloads constructor(
 ) : RecyclerView(context, attrs, defStyleAttr) {
 
 	private var onPageScrollListeners: MutableList<OnPageScrollListener>? = null
+	private val detachedViews = ArrayList<View>()
```
Comment (Koitharu) [inline app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt:17]: Is it possible to use `WeakHashMap` here? (Like in the `ActivityRecreationHandle`)

File: app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt (hunk 1 above)
Comment (VietAnh14) [reply app/src/main/kotlin/org/koitharu/kotatsu/reader/ui/pager/webtoon/WebtoonRecyclerView.kt:119]: Without it the detached view's height not changing. My guest is that request layout clear some measure cache and make the view measure again when it attach to the recyclerview again

**Review by Koitharu — APPROVED** [review]

**Pull request merged**